### PR TITLE
Feature/834 com cob offset

### PR DESCRIPTION
**Pull request by JulianHammerl**
* **Tickets addressed:** MAXGNC-834
* **Review:** By commit
* **Merge strategy:** Merge (no squash)

## Description

The cobConverter module was updated to include the center of mass (COM) offset estimation due to the phase angle correction. In addition to the unit vector pointing to the center of brightness (COB), the module now outputs another message containing the unit vector pointing to the center of mass (COM) as well as an output message containing more information about the COM computation. The noise (covariance matrix) for the COM unit vector is the same as for the COB unit vector.

The module now requires an input parameter to specify which method should be used to compute the phase angle correction ("NoCorrection" needs to be chosen to not apply any correction).

## Verification

The UnitTest was updated accordingly.

## Documentation

The documentation was updated accordingly.

## Future work

N/A

### CODE REVIEW

File: src/architecture/msgPayloadDefCpp/OpNavCOMMsgPayload.h
Comment (patkenneally) [inline src/architecture/msgPayloadDefCpp/OpNavCOMMsgPayload.h]: :small_red_triangle_down: Let's change this line to 

`Copyright (c) 2024, Laboratory for Atmospheric and Space Physics, University of Colorado at Boulder`

File: src/architecture/msgPayloadDefCpp/OpNavCOMMsgPayload.h
Comment (patkenneally) [inline src/architecture/msgPayloadDefCpp/OpNavCOMMsgPayload.h]: :small_red_triangle_down: Remove unused.

File: src/fswAlgorithms/opticalNavigation/cobConverter/cobConverter.cpp
Comment (patkenneally) [inline src/fswAlgorithms/opticalNavigation/cobConverter/cobConverter.cpp]: :large_orange_diamond: The parameter name can just be `radius`.

File: src/fswAlgorithms/opticalNavigation/cobConverter/cobConverter.cpp
Comment (patkenneally) [inline src/fswAlgorithms/opticalNavigation/cobConverter/cobConverter.cpp]: :large_orange_diamond: If you initialize `validCom` to False then you can remove your final else condition.

File: src/fswAlgorithms/opticalNavigation/cobConverter/cobConverter.cpp
Comment (patkenneally) [inline src/fswAlgorithms/opticalNavigation/cobConverter/cobConverter.cpp]: :large_orange_diamond: I realize your following the convention in the module, but I think we should refactor/move to declaring/assigning on the same line when we can. E.g. 
`Eigen::Vector3d rhat_N = cArray2EigenVector3d(ephemBuffer.r_BdyZero_N).normalized();`
or
`double rho = cArray2EigenVector3d(ephemBuffer.r_BdyZero_N).norm();`

File: src/fswAlgorithms/opticalNavigation/cobConverter/cobConverter.cpp
Comment (patkenneally) [inline src/fswAlgorithms/opticalNavigation/cobConverter/cobConverter.cpp:54]: You must know what I'm going to say about this commit ;)

File: src/fswAlgorithms/opticalNavigation/cobConverter/_UnitTest/test_cobConverter.py
Comment (patkenneally) [inline src/fswAlgorithms/opticalNavigation/cobConverter/_UnitTest/test_cobConverter.py:225]: Thank you!

**Review by patkenneally — APPROVED** [review]
:small_red_triangle_down: *Issues* to address before merge
:large_orange_diamond: *Requests* that should not block merge, but should at least be discussed
:large_blue_circle: *Recommendations* that can be ignored if desired

File: src/architecture/msgPayloadDefCpp/OpNavCOMMsgPayload.h
Comment (JulianHammerl) [reply src/architecture/msgPayloadDefCpp/OpNavCOMMsgPayload.h]: Done.

File: src/architecture/msgPayloadDefCpp/OpNavCOMMsgPayload.h
Comment (JulianHammerl) [reply src/architecture/msgPayloadDefCpp/OpNavCOMMsgPayload.h]: Done.

File: src/fswAlgorithms/opticalNavigation/cobConverter/cobConverter.cpp
Comment (JulianHammerl) [reply src/fswAlgorithms/opticalNavigation/cobConverter/cobConverter.cpp]: Done.

File: src/fswAlgorithms/opticalNavigation/cobConverter/cobConverter.cpp
Comment (JulianHammerl) [reply src/fswAlgorithms/opticalNavigation/cobConverter/cobConverter.cpp]: Done. I also made these changes to the first commit (refactor of existing module).

File: src/fswAlgorithms/opticalNavigation/cobConverter/cobConverter.cpp
Comment (JulianHammerl) [reply src/fswAlgorithms/opticalNavigation/cobConverter/cobConverter.cpp]: Done.

File: src/fswAlgorithms/opticalNavigation/cobConverter/_UnitTest/test_cobConverter.py
Comment (JulianHammerl) [reply src/fswAlgorithms/opticalNavigation/cobConverter/_UnitTest/test_cobConverter.py:225]: 🫶

File: src/fswAlgorithms/opticalNavigation/cobConverter/cobConverter.cpp
Comment (JulianHammerl) [reply src/fswAlgorithms/opticalNavigation/cobConverter/cobConverter.cpp:54]: Split the commit up into several commits.

**Pull request merged**
